src/main/java/me/gaminglounge/guiapi/Pagenation.java

```diff
@@ -3,101 +3,153 @@
 
-public class Pagenation implements InventoryHolder{
-    private MiniMessage mm;
+public class Pagenation implements InventoryHolder {
+    private final MiniMessage mm;
 
-    public Inventory inv;
-    public Component inventoryName;
-
-    public ItemStack next;
-    public ItemStack back;
-    public ItemStack close;
-
-    public ItemStack placeholderItem;
-    public int[] placeholderItemIndex;
+    private Inventory inv;
+    private Inventory returnInventory;
+    private Component inventoryName;
 
     public List<ItemStack> items;
 
-    public int currentpage;
-    public static final int numItemsOnPage =  7 * 4;
+    public Player player;
 
-    public Pagenation() {
+    public int currentpage;
+    public static final int numItemsOnPage = 7 * 4;
+
+    /*
+     * Funkitons needed:
+     * - setPlaceholderItem()
+     * - setItems()
+     */
+    public Pagenation(Player p) {
+        this.player = p;
         mm = MiniMessage.miniMessage();
-    }
 
-    public Pagenation setPlaceholderItem(ItemStack placeholderItem) {
-        this.placeholderItem = placeholderItem;
-        return this;
-    }
+        ItemBuilderManager.addBothClickEvent("GuiAPI:nextPage", (e) -> {
+            if (e.getInventory().getHolder() instanceof Pagenation pgi) {
+                e.setCancelled(true);
+                int targetPage = pgi.currentpage + 1;
+                if (targetPage < 0 || targetPage > pgi.items.size() / Pagenation.numItemsOnPage) {
+                    // error
+                    return;
+                }
+                pgi.fillPage(targetPage);
+            }
+        });
+
+        ItemBuilderManager.addBothClickEvent("GuiAPI:prevPage", (e) -> {
+            if (e.getInventory().getHolder() instanceof Pagenation pgi) {
+                e.setCancelled(true);
+                int targetPage = pgi.currentpage - 1;
+                if (targetPage < 0 || targetPage > (pgi.items.size() + Pagenation.numItemsOnPage - 1)
+                        / Pagenation.numItemsOnPage) {
+                    // error
+                    return;
+                }
+                pgi.fillPage(targetPage);
+            }
+        });
+
+        ItemBuilderManager.addBothClickEvent("GuiAPI:pagenation_return_to_old_inv_or_close", (event) -> {
+            event.setCancelled(true);
+            if (event.getInventory().getHolder() instanceof Pagenation pn) {
+                if (pn.returnInventory == null) {
+                    event.getWhoClicked().closeInventory();
+                } else {
+                    event.getWhoClicked().openInventory(pn.returnInventory);
+                }
+            }
+        });
 
-    public Pagenation setplaceholderItemIndex(int[] placeholderItemIndex) {
-        this.placeholderItemIndex = placeholderItemIndex;
-        return this;
     }
 
+    /*
+     * Sets the inventory name
+     */
     public Pagenation setInventoryName(Component inventoryName) {
```

Comment:
Should this be a javadoc comment?
```suggestion
    /**
     * Sets the inventory name
     */
    public Pagenation setInventoryName(Component inventoryName) {
```